### PR TITLE
feat(#1089): kubernetes client config with auth token

### PR DESCRIPTION
**Pull request by valisol**
- Allow configuration with new oauthToken
- Update Tests
- Update Documentation

### CODE REVIEW

File: src/manual/endpoint-kubernetes.adoc [hunk 1]
```diff
@@ -69,6 +69,8 @@ In case you need to set the client configuration explicitly on your environment
               object-mapper="objectMapper"/>
 ----
 
+*Note*: You have the option to set an `oauthToken` for the client instead of using `username` and `password`.
```
Comment (bbortt) [inline src/manual/endpoint-kubernetes.adoc:72]: I wonder if we could check if it is **either** the token **or** username and password.. a check would probably make sense; [here](https://github.com/citrusframework/citrus/blob/deac0ef52e2735ac46ce6dcc9753f34d8466a77a/connectors/citrus-kubernetes/src/main/java/org/citrusframework/kubernetes/client/KubernetesClientBuilder.java#L41).

**Comment by bbortt** [comment]
the dependency version we're using is quite old..

```xml
<k8s.client.version>1.4.34</k8s.client.version>
<k8s.model.version>1.0.65</k8s.model.version>
```

version 6 of the fabric8 client is out by now, I think. would you mind checking if the upgrade would cause big troubles? maybe we can just update the numbers without any big migrations.

I am not sure if a this-old-client is even useful with "modern" kubernetes platforms, that's why I am asking.

[link to `pom.xml`](https://github.com/citrusframework/citrus/blob/main/pom.xml#L971).

**Comment by christophd** [comment]
yes we need to update to new fabric8 client version, but unfortunately this is not an easy task as it lots of breaking changes in the client API.

File: src/manual/endpoint-kubernetes.adoc (hunk 1 above)
Comment (valisol) [reply src/manual/endpoint-kubernetes.adoc:72]: check was added to KubernetesClientConfigParser

File: connectors/citrus-kubernetes/src/main/resources/org/citrusframework/schema/citrus-kubernetes-config.xsd [hunk 2]
```diff
@@ -30,6 +30,7 @@
       <xs:attribute name="version" type="xs:string"/>
       <xs:attribute name="username" type="xs:string"/>
       <xs:attribute name="password" type="xs:string"/>
+      <xs:attribute name="oauthToken" type="xs:string"/>
```
Comment (bbortt) [inline connectors/citrus-kubernetes/src/main/resources/org/citrusframework/schema/citrus-kubernetes-config.xsd:33]: the `KubernetesClientParser` misses this property for copying.

File: connectors/citrus-kubernetes/src/main/java/org/citrusframework/kubernetes/config/annotation/KubernetesClientConfigParser.java [hunk 3]
```diff
@@ -34,38 +34,58 @@ public class KubernetesClientConfigParser implements AnnotationConfigParser<Kube
     public KubernetesClient parse(KubernetesClientConfig annotation, ReferenceResolver referenceResolver) {
         KubernetesClientBuilder builder = new KubernetesClientBuilder();
 
-        if (StringUtils.hasText(annotation.url())) {
-            builder.url(annotation.url());
-        }
+        if (isClientWithAuthToken(annotation) || isClientWithUsernameAndPassword(annotation) || isClientDefaultConfigured(annotation)) {
```
Comment (bbortt) [inline connectors/citrus-kubernetes/src/main/java/org/citrusframework/kubernetes/config/annotation/KubernetesClientConfigParser.java:37]: to keep the change smaller and the code more readable, I suggest 
```suggestion
if (!isClientWithAuthToken(annotation) && !isClientWithUsernameAndPassword(annotation) && !isClientDefaultConfigured(annotation)) {
            throw new IllegalArgumentException("Parameters not set correctly - check if either an oauthToke or password and username is set");
}
```

the, continue with the old code.

also, this sanity check does not take invocation from XML into account, as far as I see. it does only handle the annotatoin-case, that's why the parameter is the annotation itself: `KubernetesClientConfig `.

File: connectors/citrus-kubernetes/src/main/resources/org/citrusframework/schema/citrus-kubernetes-config.xsd (hunk 2 above)
Comment (valisol) [reply connectors/citrus-kubernetes/src/main/resources/org/citrusframework/schema/citrus-kubernetes-config.xsd:33]: The parser contains already the property

**Comment by bbortt** [comment]
awesome job @valisol! I've allowed myself to do some "contributor stuff" (things I wouldn't expect a first-time contributor to do), like updating all copyrights and most noteable; centralizing the rules in a `CredentialValidator`. I've then unit-tested the validation according to the following rules (see javadoc):

Validates the given credentials based on the following rules. Returns {@code true} if and only if none of the following conditions are met:
<ul>
  <li>username is set AND oauthToken is set</li>
  <li>password is set AND oauthToken is set</li>
  <li>username is <b>not</b> set AND password is set</li>
</ul>

All other combinations are valid. This makes it also compatible with the existing API, removing the need to specify an empty password.

**Review by bbortt — APPROVED** [review]

**Pull request merged**